RevenueCatUI/Templates/V2/ViewHelpers/ShadowModifier.swift

```diff
@@ -38,58 +38,44 @@ struct ShadowModifier: ViewModifier {
 
     }
 
-    var shadow: ShadowInfo
+    let shadow: ShadowInfo?
+    let shape: (any Shape)?
 
     func body(content: Content) -> some View {
-        content
-            .shadow(color: shadow.color, radius: shadow.radius, x: shadow.x, y: shadow.y)
+        if let shadow, let shape {
+            content
+                .background {
+                    GeometryReader { geometry in
+                        let rect = geometry.frame(in: .local)
+                        LayerShadowView(shape: shape,
+                                        color: shadow.color,
+                                        xOffset: shadow.x,
+                                        yOffset: shadow.y,
+                                        blur: shadow.radius * 2,
+                                        spread: 0,
+                                        rect: rect)
+                        .frame(maxWidth: .infinity, maxHeight: .infinity)
+                    }
+                }
+        } else {
+            content
+        }
     }
 }
 
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 extension View {
     func shadow(
-        shadow: ShadowModifier.ShadowInfo,
-        shape: some Shape
+        shadow: ShadowModifier.ShadowInfo?,
+        shape: (some Shape)?
     ) -> some View {
-        self.modifier(LayerShadowModifier(
-            shape: shape,
-            color: shadow.color,
-            xOffset: shadow.x,
-            yOffset: shadow.y,
-            blur: shadow.radius * 2,
-            spread: 0
+        self.modifier(ShadowModifier(
+            shadow: shadow,
+            shape: shape
         ))
     }
 }
 
-@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
-struct LayerShadowModifier: ViewModifier {
```

Comment:
The diff is a bit misleading here. `LayerShadowModifier` wasn't deleted, it was renamed to `ShadowModifier` and the original `ShadowModifier` which wasn't being used anymore was deleted.